### PR TITLE
Williamhe/1486537 add log warning and fix log message timestamp

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1486537
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -171,11 +172,13 @@ fi
         pythonBuildExitCode=${PIPESTATUS[0]}
         ELAPSED_TIME=$(($SECONDS - $START_TIME))
         echo "Done in $ELAPSED_TIME sec(s)."
+        set -e
         if [[ $pythonBuildExitCode != 0 ]]
         then
             LogWarning "${StdWarning} | Exit code: {pythonBuildExitCode} | Please review message | ${moreInformation}"
             exit $pythonBuildExitCode
         fi
+        set -e
```
Comment (daniv-msft) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:181]: Nit: do you need this one given that you already added one line 175?

File: tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs [hunk 2]
```diff
@@ -1226,13 +1226,19 @@ public void Django_CollectStaticFailure_DoesNotFailBuild()
                 CommandToExecuteOnRun = "/bin/bash",
                 CommandArguments = new[] { "-c", script }
             });
+            // Regex will match:
+            // "yyyy-mm-dd hh:mm:ss"|ERROR|Failed pip installation with exit code: 1
+            // Example:
+            // "2021-10-27 07:00:00"|ERROR|Failed to pip installation with exit code: 1
+            Regex regex = new Regex(@"""[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]) (0[0-9]|1[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])""\|WARNING\|.*");
```
Comment (daniv-msft) [inline tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs:1233]: I hope I'm not missing something, but are we only handling logs with WARNING here?

Comment (daniv-msft) [reply tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs:1233]: Specifically, given the Assert.Contains we're fixing, shouldn't we be looking for something like "'collectstatic' exited with exit code 1."?

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:181]: No, thanks for catching this :)

File: tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs (hunk 2 above)
Comment (william-msft) [reply tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs:1233]: The ask changed so the stacktrace of the failure would show; to give better reason of the failure for the user. `collectstatic exited with exit code 1` was too ambiguous. 

However I did update the regex to pattern match for `Exit code 1`

**Review by daniv-msft — APPROVED** [review]
Looks good to me!

**Review by vijaysaayi — APPROVED** [review]

**Pull request merged**